### ISSUE TITLE
Unexpected "Code is unreachable" warning for block depending on `os.name`

### ISSUE THREAD

**Issue by taldcroft** [microsoft/pyright#9774]
**Describe the bug**
Pyright labels code as unreachable if `os.name` does not match the value for the development OS. But the code may be reachable if run on a different OS.

If Pyright is trying to be strict about `os.name`, it should assume it can be any of 'posix', 'nt', 'java' (https://docs.python.org/3/library/os.html#os.name).

**Code or Screenshots**
```python
import os

if os.name == "nt":
    print("Windows")  # <-- Greyed out and marked as unreachable on Mac
else:
    print("Not windows")  # <-- Likely greyed on on a Windows machine, but I didn't check.
```
**VS Code extension or command-line**
Using pyright via Pylance version 2024.12.1 in VS code.

**Comment by erictraut** [comment]
Pyright is working as designed here.

Reachability analysis is performed based on the settings you've provided to the type checker. This includes the `pythonVersion`, `pythonPlatform`, and `defineConstant` configuration settings. If you don't provide an explicit pythonVersion or pythonPlatform setting in your configuration, pyright assumes that it should check the code based on the version and platform of the configured python interpreter.

If you'd like pyright to assume "all platforms", you can do so by setting the pythonPlatform to "all", like this:

```toml
pythonPlatform = "All"
```
